File: src/Entity/Coupon.php
<?php

namespace App\Entity;

use App\Repository\CouponRepository;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Bridge\Doctrine\Validator\Constraints\UniqueEntity;
use Symfony\Component\Validator\Constraints as Assert;
use Doctrine\Common\Collections\ArrayCollection;

/**
 * @ORM\Entity(repositoryClass=CouponRepository::class)
 *  @UniqueEntity(fields={"code"}, message="Ce code coupon existe déja !")
 */
class Coupon
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     */
    private $id;

    /**
     * @ORM\Column(type="integer", unique=true,nullable=true)
     * @Assert\Length( min=4 , max=5)
     *
     */
    private $code;

    /**
     * @ORM\Column(type="date", nullable=true)
     * @Assert\GreaterThan("today")
     * @Assert\NotBlank(message="Le champs date limite est obligatoire * ")
     */
    private $dateLimite;

    /**
     * @ORM\OneToMany(targetEntity=Commande::class, cascade={"persist", "remove"}, mappedBy="code")
     */
    protected $commande;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getCode(): ?int
    {
        return $this->code;
    }

    public function setCode(int $code): self
    {
        $this->code = $code;

        return $this;
    }

    public function getDateLimite(): ?\DateTimeInterface
    {
        return $this->dateLimite;
    }

    public function setDateLimite(\DateTimeInterface $dateLimite): self
    {
        $this->dateLimite = $dateLimite;

        return $this;
    }

    public function __construct()
    {
        $this->commande = new ArrayCollection();
    }

    public function getCommande()
    {
        return $this->answers;
    }

}

File: src/com/mycompagny/entities/Coupon.java
package com.mycompagny.entities;

public class Coupon  {

    private int id;
    private int code;
    private String dateLimite;
    private int remise;
    private int iduser;
    private String user;

    public Coupon() {
    }

    public Coupon(int code, String dateLimite, int remise) {
        this.code = code;
        this.dateLimite = dateLimite;
        this.remise = remise;
    }

    public Coupon(int id,String user, int code, String dateLimite, int remise) {
        this.id = id;
        this.user = user;
        this.code = code;
        this.dateLimite = dateLimite;
        this.remise = remise;
    }

    public Coupon(int code, String dateLimite, int remise, int iduser) {
        this.code = code;
        this.dateLimite = dateLimite;
        this.remise = remise;
        this.iduser = iduser;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getCode() {
        return code;
    }

    public void setCode(int code) {
        this.code = code;
    }

    public String getDateLimite() {
        return dateLimite;
    }

    public void setDateLimite(String dateLimite) {
        this.dateLimite = dateLimite;
    }

    public int getRemise() {
        return remise;
    }

    public void setRemise(int remise) {
        this.remise = remise;
    }

    public int getIduser() {return iduser;}

    public void setIduser(int iduser) {this.iduser = iduser;}

    public String getUser() {
        return user;
    }

    public void setUser(String user) {
        this.user = user;
    }

}
